mitlib-cf7-elements.php

```diff
@@ -75,6 +75,29 @@ function validate_dlc_filter( $result, $tag ) {
 	return $result;
 }
 
+/**
+ * Implements custom validation for radio fields.
+ *
+ * @param object $result A WPCF7_Validation object.
+ * @param object $tag    A WPCF7_FormTag object.
+ * @link https://contactform7.com/2015/03/28/custom-validation/
+ * @link https://wordpress.org/support/topic/protection-against-form-hijacking/
+ */
+function validate_options( $result, $tag ) {
+	// Check if the field value is not empty.
+	if ( ! empty( $_POST[ $tag->name ] ) ) {
+		// Look up the received value in the array of expected values.
+		$value = sanitize_text_field( wp_unslash( $_POST[ $tag->name ] ) );
```

Comment:
I don't know enough about WP or CF7 to know if this is how things are typically done, but I'd think you would want this cleaning step to persist. Otherwise you will have to repeat the same process when you go to use the value. You want to validate the data you are using.

Reply:
After looking over the source code of the email generation, this _is_ duplicate work - the contents of $_POST have already been collected and cleaned in an earlier step, but I'm not sure that I can get access to them from this function. The sample validation function that the plugin author distributes uses the technique I'm implementing here, which is to go back to $_POST and validate based on that. While this means work gets repeated, this is apparently the way that CF7 is structured?

Reply:
As long as the data you end up using has been cleaned, that's fine.

Reply:
Yep.

After confirming that there's no other issues with this change (done in Slack), I'll go ahead and merge.